pkg/events/derive/hidden_kernel_module.go

```diff
@@ -24,19 +24,21 @@ import (
 
 var (
 	foundHiddenKernModsCache *lru.Cache[uint64, struct{}]
+	eventsFromHistoryScan    *lru.Cache[*trace.Event, struct{}]
```

Comment:
I think initializing these values in the function as locals instead of globals will be easier to maintain and test.
As you are returning a function, you can initialize in it whatever values you want.